feat: implement flat dir support (#27)
ipld/js-unixfs: README.md

   Library removes indirection by taking approach similar to [multiformats][] library. Instead of passing chunker and layout config options, you pass chunker / layout / encoder interface implementations.

### Usage

You can encode a file as follows

```js
import * as UnixFS from "@ipld/unixfs"

// Create a web `TransformStream` with additional filesystem specific interface
// that allows encoding files and directories into `writable` end and reading
// IPLD blocks from `readable` end.
const archive = UnixFS.create()

// Create file writer that can be used to encode UnixFS file.
const file = UnixFS.createFileWriter(fs)
// write some content
file.write(new TextEncoder().encode("hello world"))
// Finalize file by closing it.
const { cid } = await file.close()

// close the archive to close underlying block stream.
archive.close()

// We could encode all this as car file
encodeCAR({ roots: [cid], blocks: archive.blocks })
```

> If your runtime provides [`TransforStream`][] or a [`WritableStream`][] APIs you can create filesystem writer from the `WritableStream<Block>` directly (this will allow you to control exactly how `readable`/`writable` streams deal with backpressuer)

Reviewer: ```suggestion
If your runtime provides [`TransformStream`][] or a [`WritableStream`][] APIs you can create filesystem writer from the `WritableStream<Block>` directly (this will allow you to control exactly how `readable`/`writable` streams deal with back pressure)
```

```ts
import * as UnixFS from "@ipld/unixfs"

const writeFile = async (blob: Blob, writable: WritableStream<Block>) => {
  const fs = UnixFS.createWriter({ writable })
  const file = UnixFS.createFileWriter(fs)
  file.write(new TextEncoder().encode("hello world"))
  const { cid } = await file.close()
  fs.close()

  return cid
}
```

You can encode (non sharded) directories with provided API as well

```ts
import * as UnixFS from "@ipld/unixfs"

export const demo = async () => {
  const fs = UnixFS.create()

  // write a file
  const file = UnixFS.createFileWriter(file)
  file.write(new TextEncoder().encode("hello world"))
  const fileLink = await file.close()

  // create directory and add a file we encoded above

>   const dir = UnixFS.createDirecotryWriter(fs)

Reviewer: ```suggestion
  const dir = UnixFS.createDirectoryWriter(fs)
```

  dir.write("intro.md", fileLink)
  const dirLink = await dir.close()

  // now wrap above directory with another and also add the same file
  // there

>   const root = UnixFS.createDirecotryWriter(fs)

Reviewer: ```suggestion
  const root = UnixFS.createDirectoryWriter(fs)
```
